packages/@aws-cdk/aws-servicecatalog/lib/tagoptions.ts

```diff
@@ -0,0 +1,35 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnTagOption } from '.';
+import { hashValues } from './private/util';
+import { InputValidator } from './private/validation';
+
+/**
+ * Defines a Tag Option, which are similar to tags
+ * but have multiple values per key.
+ */
+export class TagOptions {
+  /**
+  * List of CfnTagOption
+  */
+  public readonly cfnTagOptions: CfnTagOption[];
```

Comment:
We don't use properties from the L1 layer in publicly-available classes like this.

Rename the property to `tagOptions`, and make it return a `{[key: string]: string[]}` (make sure it's a defensive copy though, so none can mutate them in that way). Then, move all of the logic of creating `CfnTagOption`s into the `associateTagOptions()` method of `AssociationManager`.

Reply:
The reason why the logic is here is because tagOptions are global. Moving it to `associateTagOptions` would require us to create the L1 when an association is made and do an additional check each time to make sure no duplicate tagOptions are being created. This brings up the issue of what happens when we create one tagOption and apply it to two portfolio, it would create two tagOptions (unless we maintain a map of tagOptions as using `tryFindChild` won't work due to the different hash/portfolio).
What do you recommend Adam?

Reply:
You can still make them global by retrieving the Stack a given Portfolio belongs to with the `Stack.of(portfolio)` method.